rfc/system/0000-simplify-resources.md

```diff
@@ -0,0 +1,61 @@
+# [RFC Template] Simplify Resources
+
+**Authors:**
+
+- @granthamtaylor
+
+## 1 Executive Summary
+
+Defining resources for tasks is error prone and counter-intuitive due to the nuances around `requests` and `limits`.
+
+This RFC discusses an alternative solution to simplify the definition of resources, remove some footguns, and smoothen flytekit altogether.
+
+## 2 Motivation
+
+Some of the footguns around `requests` and `limits` are non-deterministic and not intuitive, thus can escape detection during development but have significant and detrimental impact in production.
+
+For example, the possibility of setting `requests < limits` for `mem` can lead to a pod becoming overloaded. For `cpu`, doing so can lead to CPU availability, for some multi-processing applications, to be non-deterministic, thus resulting in unpredictable performance degradation.
```

Comment:
for the memory case in particular, when a `requests != limits` the pod is considered burstable (quality of service designation) and more likely to be evicted when the node is under memory pressure